CMakeLists.txt

```
@@ -12,6 +12,19 @@ add_subdirectory(externals/)
 
 # LLVM-STUFF
 # find_package(ZSTD REQUIRED)
+
+# Check to see if were on a apple
+if(APPLE)
+
+  execute_process(
+    COMMAND brew --prefix llvm
+    OUTPUT_VARIABLE BREW_PREFIX_LLVM
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  # Mark the following LLVM_DIR to be where it's set on the Mac
+  set(LLVM_DIR "${BREW_PREFIX_LLVM}/lib/cmake/llvm" CACHE PATH "Path to LLVMConfig.cmake")
```

```suggestion
  set(LLVM_DIR "${BREW_PREFIX_LLVM}/lib/cmake/llvm" CACHE PATH "Path to LLVMConfig.cmake")
```

Only if LLVM_DIR is not set by the user could we set the LLVM_DIR for the user, i think sth like IF (NOT DEFINED LLVM_DIR) is needed before the set(...)

---

ahh gotcha, I'll make sure to see if they defined it first before changing it for them.

---

do you think i should check it with the APPLE?

instead of this:
```c++
if(APPLE)
```

we can probably do this?
```c++
if(APPLE AND LLVM_DIR NOT DEFINED)
```

---

would work yeah :)